src/Usos/Server.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace SocialiteProviders\Usos;
+
+use GuzzleHttp\Exception\BadResponseException;
+use League\OAuth1\Client\Credentials\CredentialsException;
+use League\OAuth1\Client\Credentials\TemporaryCredentials;
+use League\OAuth1\Client\Credentials\TokenCredentials;
+use SocialiteProviders\Manager\OAuth1\Server as BaseServer;
+use SocialiteProviders\Manager\OAuth1\User;
+
+class Server extends BaseServer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function urlTemporaryCredentials()
+    {
+        return $this->getDomain().'/services/oauth/request_token';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function urlAuthorization()
+    {
+        return $this->getDomain().'/services/oauth/authorize';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function urlTokenCredentials()
+    {
+        return $this->getDomain().'/services/oauth/access_token';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function urlUserDetails()
+    {
+        return $this->getDomain().'/services/users/user'.'?fields='.$this->getFieldsSelector();
+    }
+
+    private function getDomain()
+    {
+        return config('services.usos.domain');
+    }
+
+    private function getFieldsSelector()
+    {
+        return config('services.usos.profile_fields_selector', 'id|first_name|last_name|email|photo_urls');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function userDetails($data, TokenCredentials $tokenCredentials)
+    {
+        $user = new User();
+        $user->id = $data['id'];
+        $user->nickname = $data['first_name'].' '.$data['last_name'];
+        $user->name = $data['first_name'].' '.$data['last_name'];
+        $user->avatar = $data['photo_urls']['50x50'] ?? null;
+        $user->email = $data['email'] ?? null;
+
+        $used = ['id', 'first_name', 'last_name', 'email'];
+
+        $user->extra = array_diff_key($data, array_flip($used));
```

Comment:
I don't think we need this to filter the keys. Just use `setRaw` to set all the keys?